src/sagemaker_xgboost_container/algorithm_mode/serve_utils.py

```diff
@@ -0,0 +1,254 @@
+    :param keys: strings denoting selected keys (list)
+    :param objective: objective xgboost training function (str)
+    :param num_class: number of classes for multiclass classification (str, optional)
+    :return: selected content (list of dict)
+    """
+    if objective not in VALID_OBJECTIVES:
+        raise ValueError("Objective `{}` unsupported for selectable inference content.".format(objective))
+
+    if not set(keys).issubset(VALID_OBJECTIVES[objective]):
+        logging.warning("Selected key(s) {} incompatible for objective '{}'. "
+                        "Please use list of compatible selectable inference content: {}"
+                        .format(set(keys).difference(VALID_OBJECTIVES[objective]), objective,
+                                VALID_OBJECTIVES[objective]))
+
+    content = []
+    for prediction in data:
+        output = {}
+        if PREDICTED_LABEL in keys:
+            output[PREDICTED_LABEL] = _get_predicted_label(objective, prediction)
+        if LABELS in keys:
+            output[LABELS] = _get_labels(objective, num_class=num_class)
+        if PROBABILITY in keys:
+            output[PROBABILITY] = _get_probability(objective, prediction)
+        if PROBABILITIES in keys:
+            output[PROBABILITIES] = _get_probabilities(objective, prediction)
+        if RAW_SCORE in keys:
+            output[RAW_SCORE] = _get_raw_score(objective, prediction)
+        if RAW_SCORES in keys:
+            output[RAW_SCORES] = _get_raw_scores(objective, prediction)
+        if PREDICTED_SCORE in keys:
+            output[PREDICTED_SCORE] = prediction
+        content.append(output)
+    return content
+
+
+def _encode_selected_content_csv(content, ordered_keys_list):
+    """Encode content is csv format with the ordered_keys_list as the header.
+
+    :param content: list of selected content
+    :param ordered_keys_list: list of selected content keys
+    :return: csv string
+    """
+    sio = io.StringIO()
+    cw = csv.writer(sio)
+    for row in content:
+        csv_row = [row[key] for key in ordered_keys_list]
+        cw.writerow(csv_row)
+    return sio.getvalue()
+
+
+def _write_record(record, key, value):
+    record.label[key].float32_tensor.values.extend(value)
+
+
+def _encode_selected_content_recordio_protobuf(content):
+    """Encode list of dictionaries into recordio protobuf format. Dictionary keys are "label" keys.
+
+    :param content: list of dictionaries
+    :return: recordio bytes
+    """
+    record_bio = io.BytesIO()
+    recordio_bio = io.BytesIO()
+    record = Record()
+    for item in content:
+        for key in item.keys():
+            value = item[key] if type(item[key]) is list else [item[key]]
+            _write_record(record, key, value)
+        record_bio.write(record.SerializeToString())
+        record.Clear()
+        _write_recordio(recordio_bio, record_bio.getvalue())
+    return recordio_bio.getvalue()
+
+
+def encode_selected_content(content, keys, accept):
```

Comment:
We also have encoder code that should be useable by customers during script mode, please move this to that submodule.

Reply:
These encoders are a little niche because they're supposed to be used specifically with the layout of the `content` argument. I think it makes more sense to keep them in the `serve_utils` submodule